src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionSettings.cs

```diff
@@ -48,6 +52,8 @@ internal sealed class HttpConnectionSettings
 
         internal IDictionary<string, object> _properties;
 
+        internal Func<string, int, CancellationToken, ValueTask<Stream>> _customConnect = ConnectHelper.ConnectAsync;
```

Comment:
Nit: it could be argued that the `string` and `int` parameters of the delegate do not clearly communicate meaning. Any reason why we can't use a custom delegate here?

Reply:
We can and I agree, but @stephentoub was against it and I didn't think it is a huge enough benefit to fight for it.

I think that adding a second host/port pair for proxy would be really confusing though, so we will need either a new delegate type or a new `EventArgs`-like type which we'd need to allocate.

Reply:
My preference would be to have something like:
```C#
Func<HttpRequestMessage, CancellationToken, ValueTask<Stream>>
```
and I don't think a custom delegate adds anything meaningful there.  If we end up just taking string/int, or worse multiple string/int pairs, then a custom delegate is fine.

Reply:
One option I've been thinking of tonight is to give `HttpConnectionPool` an informational base class so that we can pass it directly to the callback. No allocations that way, and it gives us an option to expose more in the future if we need to.

I don't think `HttpRequestMessage` makes sense when multiple requests can be sent on a single connection. Granted this is an API for advanced usage so I think we do have some leeway here, but it'd be very easy to assume the wrong behavior.